### DISCUSSION TITLE
Move away from Lerna?

### DISCUSSION THREAD

**Discussion by jamonholmgren** [mobxjs/mobx-state-tree#1600] [Ideas] [answered]
I'm taking over as lead maintainer of MobX-State-Tree, and one thing I'd like to do is simplify the maintenance of MST in general.

In most of my open source libraries, I've gone away from Lerna and had individual projects for every repo, released with [semantic-release](https://github.com/semantic-release/semantic-release). This has worked pretty well overall.

As far as I can tell, Lerna isn't doing a lot for us here other than holding some rather outdated examples and mst-middleware. And it's getting in the way as I start working within the repo in the now-familiar ways that Lerna gets in the way (namely not playing well with tooling like semantic-release).

My proposal:

1. Make `./packages/mobx-state-tree` the main repo
2. Remove Lerna
3. Archive the examples
4. Move `mst-middleware` to its own repo
5. Add semantic-release to `mobx-state-tree` and `mst-middleware` to make it simple to maintain and release new versions

Does anyone have objections to this?

cc @mweststrate

**Comment by jamonholmgren** [comment]
I intend to merge as many PRs as make sense before doing this, as they'll all be conflicted like crazy once we do.

**Comment by danielkcz** [comment]
Hey Jamon. As you might have noticed we have started using https://github.com/atlassian/changesets in MobX monorepo and it's a great and simple system. In my opinion, it's superior to `semantic-release` as it gives better control over what and when to publish. I especially love those [automated changelogs](https://github.com/mobxjs/mobx/pull/2616) mentioning the author automatically.

Lerna is indeed a bit of an outdated tool. The changesets are a great replacement for it as well.

In a long run, once all issues and PRs are resolved here, it might be worth considering moving these packages to `mobx` monorepo as well to have it all consolidated.

**Reply by jamonholmgren** [reply]
Thanks @FredyC, appreciate the pointer to changesets!

I have a similar setup with semantic-release on my other repos (also mentioning the author in [automated changlogs](https://github.com/infinitered/ignite/releases)), and with good control over when to publish. So overall I've been pretty happy with semantic-release, but happy to learn more about alternatives and what they offer. I'm meeting with the creator of [Auto](https://intuit.github.io/auto/) soon as he mentioned some interesting benefits of using that system. In the end, I may end up staying with semantic-release since that's what I know the best and to keep my release process consistent across all the different repos I use, but like I said I'm keeping an open mind for now. 😁

**Reply by danielkcz** [reply]
Sure, if you are comfortable with it, it's fine.

And what do you think about eventually moving the repo to our monorepo? You could still be responsible for issues/PRs toward it there of course.

**Comment by mweststrate** [comment, marked as answer]
If memory serves me well Lerna served two purposes:

1. make sure example projects run / link against latest source
2. coordinate mst-middleware with mst releases

I think 2 isn't really needed anymore, especially if releases are automated.

For 1, this can be solved with less obtrusive solutions than lerna, like workspaces, relative-deps or linking in post install scripts. Or even not linking automatically at all; it is a bit less urgent now the library is in a mature state 

So yeah, feel free to drop Lerna :)

**Comment by mweststrate** [comment]
I think that would be quite confusing in the issue tracker? I noticed mobx
and mobx-react tend to blend much more together into one thing in peoples
mind, while a lot of mobx specific problems / questions dont apply to MST
and vice versa

**Comment by coolsoftwaretyler** [comment]
Done and done. Thanks all!
